packages/bot-web-ui/src/stores/run-panel-store.js

```diff
@@ -4,23 +4,82 @@ import { error_types, unrecoverable_errors, observer, message_types } from '@der
+    total_stake: 0,
+    won_contracts: 0,
+};
+let get_statistics;
+
 export default class RunPanelStore {
     constructor(root_store) {
         this.root_store = root_store;
         this.dbot = this.root_store.dbot;
         this.registerCoreReactions();
+
+        get_statistics = JSON.parse(sessionStorage.getItem(statistics_storage_key));
+
+        reaction(
+            () => this.statistics,
+            statistics => {
+                const { client } = this.root_store.core;
+                const stored_statistics = JSON.parse(sessionStorage.getItem(statistics_storage_key)) ?? {};
+
+                const new_statistics = { statistic: statistics };
+                stored_statistics[client.loginid] = new_statistics;
+
+                sessionStorage.setItem(statistics_storage_key, JSON.stringify(stored_statistics));
+            }
+        );
+
+        reaction(
+            () => this.root_store.core.client.loginid,
+            () => {
+                const { core, transactions, journal, contract_card } = this.root_store;
+                const { client } = core;
+
+                if (client.is_logged_in) {
+                    const statictics = JSON.parse(sessionStorage.getItem(statistics_storage_key));
+                    const journal_messages = JSON.parse(sessionStorage.getItem('journal_cache'));
+                    const transaction_elements = JSON.parse(sessionStorage.getItem('transaction_cache'));
+
+                    this.statistics =
+                        statictics && statictics[client.loginid]
+                            ? statictics[client.loginid].statistic
+                            : empty_statistic;
+
+                    journal.unfiltered_messages =
+                        journal_messages && journal_messages[client.loginid]
+                            ? journal_messages[client.loginid].journal_message
+                            : [];
+                    transactions.elements =
+                        transaction_elements && transaction_elements[client.loginid]
+                            ? transaction_elements[client.loginid].transaction_element
+                            : [];
+                } else {
+                    this.statistics = empty_statistic;
+                    journal.unfiltered_messages = [];
+                    transactions.elements = [];
+                }
+
+                this.is_running = false;
+                this.has_open_contract = false;
+                contract_card.clear();
+                this.setContractStage(contract_stages.NOT_RUNNING);
+            }
+        );
     }
 
     run_id = '';
 
-    @observable statistics = {
-        lost_contracts: 0,
-        number_of_runs: 0,
-        total_profit: 0,
-        total_payout: 0,
-        total_stake: 0,
-        won_contracts: 0,
-    };
+    @observable statistics = get_statistics
+        ? get_statistics[this.root_store.core.client.loginid]?.statistic
+        : empty_statistic;
```

Comment:
Can be shortened as per below:

```suggestion
@observable statistics = get_statistics?.[this.root_store.core.client.loginid]?.statistic ?? empty_statistic;
```